### PR TITLE
[Stable] Fix issues with type inference and augment tests

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

**Comment by kasper-piskorski** [comment]
!rtg - ai.grakn.factory.GraknTxJanusTest.ai.grakn.factory.GraknTxJanusTest

**Comment by kasper-piskorski** [comment]
no love for this one ;p?

**Comment by fppt** [comment]
Sorry Kasper been out of my "groove" of doing reviews in the mornings. Will show love now :)

File: grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/utils/conversion/RoleTypeConverter.java [hunk 1]
```diff
@@ -32,12 +37,21 @@
...
+    public Multimap<RelationshipType, Role> toRelationshipMultimap(Role entryRole) {
         Multimap<RelationshipType, Role> relationMap = HashMultimap.create();
-        role.subs().forEach(roleType -> {
-                    roleType.relationshipTypes()
+
+        Set<Role> roleHierarchy = Sets.newHashSet(entryRole);
+        if (Schema.MetaSchema.isMetaLabel(entryRole.getLabel())) {
+            entryRole.subs().forEach(roleHierarchy::add);
+        } else {
+            ReasonerUtils.supers(entryRole).stream().map(Concept::asRole).forEach(roleHierarchy::add);
```
[the remark this answers is not in the document]
Comment (kasper-piskorski) [reply grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/utils/conversion/RoleTypeConverter.java]: I absolutely agree :)

**Pull request merged**